### ISSUE TITLE
no-descending-specificity reports in cross-regions

### ISSUE THREAD

**Issue by Yash-Singh1** [stylelint/stylelint#5111]
## Bug

> Clearly describe the bug

Let's take the following example:

```css
/* #region Icons in the toolbar */

img#a   {
  width: 10%;
}

img#b {
  width: 5%;
}

/* #endregion */

/* #region Remove Image Borders */

img /* error */ {
  border: none;
  outline: none;
}

/* #endregion */
```

Of course, the above example is not the best use of [`vscode`'s region feature](https://code.visualstudio.com/docs/languages/css#_folding), but it is just an example.

Where you see `/* error */` is where an error expecting `img` to come before the `img#a` and `img#b` is. This error should not be reported because the  `img#a` and `img#b` are in another region.

> Which rule, if any, is the bug related to?

`no-descending-specificity`

> What code is needed to reproduce the bug?

```css
/* #region Icons in the toolbar */

img#a   {
  width: 10%;
}

img#b {
  width: 5%;
}

/* #endregion */

/* #region Remove Image Borders */

img /* error */ {
  border: none;
  outline: none;
}

/* #endregion */
```

> What stylelint configuration is needed to reproduce the bug?

```json
{
  "extends": "stylelint-config-standard"
}
```

Remember to have `stylelint-config-standard` installed

> Which version of stylelint are you using?

From `package.json`:

```json
{
  "devDependencies": {
      "stylelint": "^13.8.0",
      "stylelint-config-standard": "^20.0.0"
  }
}
```

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

Visual Studio Code plugin

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

Nope, this is about CSS.

> What did you expect to happen?

No warnings to be flagged

> What actually happened (e.g. what warnings or errors did you get)?

Warnings above

## Links of Reference

If you don't know about regions folding in VSCode, here is a reference: https://code.visualstudio.com/docs/languages/css#_folding

**Comment by m-allanson** [comment]
Hey @Yash-Singh1, thank you for the nice and clear write up. However, I think your example demonstrates a 'correct' rule violation.

When rendering your CSS in a browser, the folding or unfolding of regions won't make any difference to how the page is displayed. The `/* #region */` comments will be ignored by the browser, and your rules will be applied in the order that they're written.  

You can allow your example pattern in your CSS by switching off the rule, or by using [stylelint's ignore comments](https://stylelint.io/user-guide/ignore-code).

**Comment by jeddy3** [comment]
> When rendering your CSS in a browser, the folding or unfolding of regions won't make any difference to how the page is displayed. The `/* #region */` comments will be ignored by the browser, and your rules will be applied in the order that they're written.

That's correct. The intent of the rule is to align the source order with specificity to avoid unexpected gotchas. As regions will be ignored by the browser, the rule is suggestions your order your rules (and your regions) like so:

```css
/* #region Remove Image Borders */

img /* error */ {
  border: none;
  outline: none;
}

/* #endregion */

/* #region Icons in the toolbar */

img#a   {
  width: 10%;
}

img#b {
  width: 5%;
}

/* #endregion */

```

As @m-allanson said, thanks for the clear write-up (and for the introduction to VS Code regions that I didn't know about), but I'll close the issue as it's behaving as designed.
